src/Enforcers/ValidationEnforcer.php

```diff
@@ -0,0 +1,181 @@
+            {
+                continue;
+            }
+
+            //Attempt to validate each property that is on the object.
+            foreach ($prop['validators'] as $validator)
+            {
+                if (!$validator->validate($arr[$propName]))
+                {
+                    $validationErrors->addError($validator->getErrorMessage($propName), $propName);
+                }
+            }
+        }
+
+        if (count($validationErrors->getErrors()))
+        {
+            throw $validationErrors;
+        }
+    }
+
+    /**
+     * Returns an associative array with key/value pairs as follows:
+     *   key: a property fo the provided class
+     *   value: an associative array containing two keys; a 'reflectionProperty' which
+     *     is an object of type ReflectionProperty so some metadata about the property
+     *     can be referenced in other functions (such as nullability of the property)
+     *     and 'validators' which is an array of instantiated Validators for every
+     *     validation that the given property has defined on it.
+     *
+     * @param string className - The name of the class for which the validation metadata
+     * is to be created.
+     * @return array See above description.
+     */
+    public function getValidationMetadata(string $className): array
+    {
+        $validators = [
+            "int"      => new IntValidator(),
+            "string"   => new StringValidator(),
+            "bool"     => new BoolValidator(),
+            "DateTime" => new DateValidator(),
+            "notNull"  => new NotNullValidator(),
+        ];
+
+        $metadata = [];
+
+        $ref   = new ReflectionClass($className);
+        $props = $ref->getProperties();
+
+        //For each property defined on the class.
+        foreach($props as $prop)
+        {
+            $propName    = $prop->getName();
+            $propType    = $prop->getType()->getName();
+            $propAttrs   = $prop->getAttributes();
+            $propCanNull = $prop->getType()->allowsNull();
+
+            $metadata[$propName] = ['reflectionProperty' => $prop, 'validators' => []];
+            $validator = array_key_exists($propType, $validators) ? $validators[$propType] : null;
+
+            if (!is_null($validator))
+            {
+                $metadata[$propName]['validators'][] = $validator;
+            }
+
+            if (!$propCanNull)
+            {
+                $metadata[$propName]['validators'][] = $validators['notNull'];
+            }
+
+            //If the property has any validation attributes make sure they are included.
+            foreach($propAttrs as $attr)
+            {
+                $validatorName = $attr->getName();
+                if (is_subclass_of($validatorName, Validator::class))
+                {
+                    $attrValidator = new $validatorName(...$attr->getArguments());
+                    $metadata[$propName]['validators'][] = $attrValidator;
```

Comment:
You can use `$metadata[$propName]['validators'][] = $attr->newInstance()` which will create the Validator and supply the arguments.